src/minidom_utils/writer.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright (C) 2017 Kisio Digital and/or its affiliates.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU Affero General Public License as published by the
+// Free Software Foundation, version 3.
+
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+// details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>
+
+//! Writer to serialize an `Element` (its name, attributes and childrens)
+
+use crate::Result;
+use minidom::{Element, Node};
+use quick_xml::{
+    events::{BytesDecl, BytesEnd, BytesStart, BytesText, Event},
+    Writer,
+};
+use std::io::Write;
+const XML_VERSION: &str = "1.0";
+const ENCODING: &str = "UTF-8";
+
+pub struct ElementWriter {
+    element: Element,
+    indent: bool,
+}
+
+// Publicly exposed methods
```

Comment:
I added comments about why I separated into two `impl` sections in 2ab9adc6427eedffae0a609eac4b2ddb3d8666dc.